playkit/src/main/java/com/kaltura/playkit/player/TrackSelectionHelper.java

```diff
@@ -323,21 +324,30 @@ private String getExternalSubtitleLanguage(Format format) {
         }
     }
 
-    private boolean ignoreTextTrackOnPreference(Format format) {
+    private boolean discardTextTrackOnPreference(Format format) {
+        PKSubtitlePreference subtitlePreference = playerSettings.getSubtitlePreference();
+        if (subtitlePreference == PKSubtitlePreference.OFF) {
+            return false;
+        }
+
         String languageName = format.language;
         boolean isExternalSubtitle = isExternalSubtitle(format);
-        boolean isPreferInternalSubtitles = playerSettings.isPreferInternalSubtitles();
 
         if (isExternalSubtitle) {
             languageName = getExternalSubtitleLanguage(format);
+            if (subtitlePreference == PKSubtitlePreference.INTERNAL && !subtitleListMap.get(languageName).containsKey(languageName)) {
```

Comment:
Check null first `subtitleListMap.get(languageName)` before checking for contains.

Reply:
Done